BeautifulSoupScript_edits.py

```diff
@@ -0,0 +1,213 @@
+        with open("data_file.json", "w") as write_file:
+            json.dump(download_dict, write_file)
+        '''
+#Parsing the journal and miscelleanous publications for metadata at the bottom of webpage
+def parse_Pub_Jour(pub, journal_key):
+    pubLinks = pub.find_all('a')
+    #Link should only be the link to the publication webpage
+    detailLink = pubLinks[0].get('href', None)
+
+    detailURL = '{}{}'.format(OIC_ROOT, detailLink)
+
+    print(detailURL)
+
+    try:
+        pubElem = fetchPageHTML(detailURL)
+    except ConnectionError or HTTPError:
+        return None
+    
+    pubSoup = BeautifulSoup(pubElem.text, 'lxml')
+
+    pubContainer = pubSoup.find(class_='content-inner-left')
+
+    if pubContainer is None:
+        return None
+    #Metadata list is the last unordered list on the webpage
+    dataList = pubContainer.find_all('ul')[-1]
+
+    metadata = dataList.find_all('li')
+
+    #CHDS metadata dictionary
+    if journal_key == 2:
+        if metadata[2].text[0:4] == 'ISBN':
+            return {
+            'series': metadata[0].text,
+            'publicationInfo': metadata[1].text,
+            'isbnNumber': metadata[2].text,
+            'extent': metadata[3].text
+            }
+        else:
+            return {
+            'series': metadata[1].text,
+            'publicationInfo': metadata[2].text,
+            'isbnNumber': metadata[3].text,
+            'extent': metadata[4].text
+            }
+    #OIDA metadata dictionary
+    elif journal_key == 7:
+        return {
+        'series': metadata[0].text,
+        'publicationInfo': metadata[1].text,
+        'extent': metadata[2].text
+        }
+    #OIMP and OIS metadata dictionary
+    elif journal_key == 8  or journal_key == 10:
+        if len(metadata) == 2 or len(metadata) == 3:
+            return {
+                'publicationInfo': metadata[0],
+                'extent': metadata[1]
+            }
+        elif len(metadata) == 4: #OIP 117 has metadata in middle of page instead of at the bottom
+            return None
+        else:
+            return {
+                'series': metadata[0].text,
+                'publicationInfo': metadata[1].text,
+                'isbnNumber(hardbook)': metadata[2].text,
+                'isbnNumber(ebook)': metadata[3].text,
+                'extent': metadata[4].text
+            }
+
+    #OIS metadata dictionary(Sometimes series firsts and sometimes it's last in the list)
+    elif journal_key == 11:
+        return {
+            'series': metadata[3].text,
+            'publicationInfo': metadata[0].text,
+            'isbnNumber': metadata[1].text,
+            'extent': metadata[2].text
+            }
+    #Miscellaneous metadata dictionary
+    elif journal_key == 13:
+        return {
+            'title': metadata[0].text,
+            'publicationInfo': metadata[1].text,
+            'isbnNumber-13': metadata[3].text,
+            'isbnNumber-10': metadata[4].text,
```

Comment:
Same comment as above, I'd merge these two things! (Thanks for the attention to detail in capturing this though!)